vault/mount.go

```diff
@@ -1593,7 +1593,18 @@ func (c *Core) setupMounts(ctx context.Context) error {
 
 		// Ensure the path is tainted if set in the mount table
 		if entry.Tainted {
-			c.router.Taint(ctx, entry.Path)
+			// Calculate any namespace prefixes here, because when Taint() is called, there won't be
+			// a namespace to pull from the context. This is similar to what we do above in c.router.Mount().
+			path := entry.Namespace().Path + entry.Path
+			if entry.Path != path {
+				nsfc, err := namespace.FromContext(ctx)
+				if err != nil {
+					c.logger.Error("error when trying to get the namespace from the context", "error", err)
+				} else {
+					c.logger.Debug("tainting a mount but path and namespaced path disagree", "entry_path", entry.Path, "entry_namespace_path", entry.Namespace().Path, "namespace_from_context", nsfc)
```

Comment:
I don't understand the nature of what we're logging here. Why are the paths in disagreement?

If I'm reading this code right, this if branch will be taken any time we're in a namespace that's not the root namespace. So if I'm in namespace `foo`, and we're calling `Taint()` on an entry that has path `/bar`, `entry.Path` will be `/bar` and `entry.Namespace().Path` will be `/foo` and the resulting path that will be tainted is `/foo/bar`, which is as it should be.

Were we to not do the path calculation that we do on line 1598, and simply call `Taint()` as before, the namespace wouldn't be present in the context when we try to pull it out, so the resulting path we would taint would be `/bar`, which is not correct, given that we're in the `foo` namespace.

So I'm confused about why the path and the namespaced path disagree.

Reply:
Yeah, you're right.

I took this from the change we made to resolve the issue, and I think I 'trusted' my memory a little. Good catch, and I think I'll just adjust this to log before we call `Taint`.

Thanks for this!